cloudbuild.yaml

```diff
@@ -0,0 +1,107 @@
+# Parameterized Cloud Build Configuration
+# This file defines a multi-step build process including:
+# - Building a temporary application image.
+# - Running database migrations using the temporary image and Cloud SQL Proxy.
+# - Copying migration files to the workspace.
+# - Building a final application image tagged with the commit SHA.
+# - Pushing the final image to Artifact Registry.
+# - Committing generated migration files back to a Git repository.
+
+steps:
+  - name: gcr.io/cloud-builders/curl
+    id: download-proxy
+    waitFor: ['-'] # Start immediately
+    args:
+      - '-o'
+      - cloud-sql-proxy
+      - 'https://storage.googleapis.com/cloud-sql-connectors/cloud-sql-proxy/${_PROXY_VERSION}/cloud-sql-proxy.linux.amd64'
+  - name: gcr.io/cloud-builders/gcloud
+    id: chmod-proxy
+    waitFor: ['download-proxy']
+    args:
+      - '-c'
+      - chmod +x cloud-sql-proxy; ls; pwd; ./cloud-sql-proxy -h;
+    entrypoint: bash
+  - name: gcr.io/cloud-builders/docker
+    id: build-temp-image
+    waitFor: ['-'] # Can start alongside proxy download/chmod
+    args:
+      - build
+      - '-t'
+      - '${_ARTIFACT_REGISTRY_HOST}/${_ARTIFACT_REGISTRY_PROJECT_ID}/${_ARTIFACT_REGISTRY_REPO}/${_APP_IMAGE_NAME}:temp'
+      - .
+  - name: '${_ARTIFACT_REGISTRY_HOST}/${_ARTIFACT_REGISTRY_PROJECT_ID}/${_ARTIFACT_REGISTRY_REPO}/${_APP_IMAGE_NAME}:temp'
+    id: run-migrations
+    waitFor: ['build-temp-image', 'chmod-proxy']
+    env:
+      - 'DB_HOST=${_DB_HOST}'
+      - 'DB_NAME=${_DATABASE_NAME}'
+      - 'DB_USER=${_DATABASE_USER}'
+      - 'DB_PORT=${_DATABASE_PORT}'
+      - 'INSTANCE_CONNECTION_NAME=${_INSTANCE_CONNECTION_NAME}'
+    args:
```

Comment:
just want to point out that these can use "shell" instead of "args" so you can write a full multiline shell script here.  makes some of this a lot easier, no need for semicolons, more readable, etc